src/coreclr/debug/daccess/enummem.cpp

```diff
@@ -533,7 +533,7 @@ HRESULT ClrDataAccess::DumpManagedExcepObject(CLRDataEnumMemoryFlags flags, OBJE
     DumpManagedExcepObject(flags, exceptRef->GetInnerException());
 
     // Dump the stack trace array object and its underlying type
-    I1ARRAYREF stackTraceArrayObj = exceptRef->GetStackTraceArrayObject();
+    OBJECTREF stackTraceArrayObj = exceptRef->GetStackTraceArrayObject();
```

Comment:
When capturing a minidump or triage dump, we need to ensure that the underlying stack trace objects make their way into the dump file.  The code in enummem.cpp only dump the first level object, which is now just the `object[]` which contains the pointer to the `byte[]` we care about.

So I think this logic needs to be expanded so that everything within `if (stackTraceArrayObj != NULL)` (following the above line) is duplicated for the first array element in stackTraceArrayObj.

I don't think we need to capture the extra entries in the object[] for a minidump, but if those entries would be useful in a minidump scenario, please consider adding those too.

Reply:
@leculver thank you for spotting this. I will make sure the `byte[]` array from the `object[]` is dumped too. As for dumping the rest of the `object[]` for minidump, we were not dumping the `_dynamicMethods` array before either (that was where those entries were before my change) and I also don't see them to be useful in minidumps.